Framework/Kernel/src/PropertyManager.cpp

```diff
@@ -173,6 +174,43 @@ void PropertyManager::filterByProperty(const Kernel::TimeSeriesProperty<bool> &f
   }
 }
 
+/**
+ * Create a partial copy of this object such that every time series property is cloned according to the input TimeROI.
+ * A partially cloned time series property should include all time values enclosed by the ROI regions,
+ * each defined as [roi_start,roi_end), plus the values immediately before and after an ROI region, if available.
+ * Properties that are not time series will be cloned with no changes.
+ * @param timeROI :: a series of time regions used to determine which time series values should be included in the copy.
+ */
+PropertyManager *PropertyManager::cloneInTimeROI(const Kernel::TimeROI &timeROI) {
+  PropertyManager *newMgr = new PropertyManager();
+  newMgr->m_orderedProperties.reserve(m_orderedProperties.size());
+  // We need to do a deep copy of the property pointers here
+  for (auto prop : m_orderedProperties) {
+    auto tsp = dynamic_cast<ITimeSeriesProperty *>(prop);
+    std::unique_ptr<Property> newProp;
+    if (tsp)
+      newProp = std::unique_ptr<Property>(tsp->cloneInTimeROI(timeROI));
```

Comment:
I don't recall the difference between `std::unique_ptr` constructor and `std::make_unique`. Do you know if this is creating a temporary second copy?